bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/BKException.java

```diff
@@ -271,6 +273,13 @@ public static String getMessage(int code) {
          */
         int LedgerIdOverflowException = -106;
 
+        /**
+         * Failure to serialize/deserialize.
+         *
+         * @since 4.9
+         */
+        int SerializationException = -107;
```

Comment:
SerializationException seems to be a very generic exception. Shall we call it a "MetadataCorruptedException" indicating it is a corrupted metadata?

Reply:
renamed to metadata serialization exception. it's not for corrupted metadata. This exception occurs when there's and exception in the call to #serialize, which can really only be an IOException when doing the protobuf binary encoding, which should never happen as the IO is all on memory.